app/assets/javascripts/controllers/flavor_credentials/flavor_form_controller.js

```diff
@@ -0,0 +1,105 @@
+ManageIQ.angular.app.controller('flavorFormController', ['$scope', '$http', 'flavorId', 'miqService', 'API', function($scope, $http, flavorId, miqService, API) {
+  var vm = this;
+
+  var init = function() {
+    vm.afterGet = false;
+
+    vm.flavorModel = {
+      name: '',
+      ram: '',
+      vcpus: '',
+      disk: '',
+      swap: '',
+      rxtx_factor: '',
+      ems_id: '',
+    };
+
+    vm.model = 'flavorModel';
+    vm.ems_list = [];
+
+    $scope.newRecord = flavorId === 'new';
+
+    ManageIQ.angular.scope = vm;
+
+    $http.get('/flavor/ems_list')
+      .then(getEmsFormDataComplete)
+      .catch(miqService.handleFailure);
+
+    API.get('/api/providers?collection_class=ManageIQ::Providers::EmbeddedAutomationManager')
+      .then(getManagerResource)
+      .catch(miqService.handleFailure);
+  };
+
```

Comment:
Reset button is enabled when record is not new, flavor updating is not possible, for now I don`t need it.